### PR TITLE
Feature job pooldata model

### PR DESCRIPTION
**Pull request by salguerod**
Enhancements to the JobPooling Feature including:
1) A basic Python command line tool to automatically create/provision 25 pooled STS jobs per input bucket and register/sync them with SDRS via the REST API, 2) Java server side business logic validation of that sts-to-sdrs registration/provisioning operation, and 3)improved logging functionality.

### CODE REVIEW

File: scripts/provisioning/command_line.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+import argparse
+import datetime
+import json
+import logging
+import requests
+
+import googleapiclient.discovery
+
+LOGGER = logging.getLogger('sdrs_provisioning_cli')
```
Comment (eshen1991) [inline scripts/provisioning/command_line.py]: consider to add handler to log to a file

File: src/main/java/com/google/gcs/sdrs/controller/JobPoolController.java [hunk 2]
```diff
@@ -40,8 +40,7 @@
 @Path("/stsjobpool")
 public class JobPoolController extends BaseController {
 
-  JobPoolService jobPoolService =
-      new JobPoolServiceImpl(); // TODO need to wire up the instantiation via a factory
+  private JobPoolService jobPoolService = JobPoolServiceImpl.getInstance(); // TODO need to wire up the instantiation via a factory
```
Comment (eshen1991) [inline src/main/java/com/google/gcs/sdrs/controller/JobPoolController.java]: is the TODO done?

File: src/main/java/com/google/gcs/sdrs/service/impl/JobPoolServiceImpl.java [hunk 3]
```diff
@@ -40,12 +73,66 @@ public Integer createJob(PooledJobCreateRequest request) {
...
+    try {
+      TransferJob transferJob =
+          StsUtil.getExistingJob(
+              client, pooledJobCreateRequest.getProjectId(), pooledJobCreateRequest.getName());
+      if (transferJob == null || transferJob.isEmpty()) {
+        return false;
+      }
+      return true;
+    } catch (IOException e) {
+      e.printStackTrace();
```
Comment (eshen1991) [inline src/main/java/com/google/gcs/sdrs/service/impl/JobPoolServiceImpl.java]: consider using RetentionUtil.convertStackTrace. https://github.com/GoogleCloudPlatform/storage-sdrs/blob/develop/src/main/java/com/google/gcs/sdrs/service/worker/rule/impl/StsRuleExecutor.java#L201-L206 as e.printStackTrace() creates multiple log entries in StackDriver

**Comment by salguerod** [comment]
Will reopen pull request with added enhancements to feature

**Pull request closed without merging**